Index: llvm-toolchain-19/clang-tools-extra/clangd/index/remote/monitor/Monitor.cpp
===================================================================
--- llvm-toolchain-19.orig/clang-tools-extra/clangd/index/remote/monitor/Monitor.cpp
+++ llvm-toolchain-19/clang-tools-extra/clangd/index/remote/monitor/Monitor.cpp
@@ -61,7 +61,6 @@ int main(int argc, char *argv[]) {
   std::string Output;
   google::protobuf::util::JsonPrintOptions Options;
   Options.add_whitespace = true;
-  Options.always_print_primitive_fields = true;
   Options.preserve_proto_field_names = true;
   const auto JsonStatus =
       google::protobuf::util::MessageToJsonString(Response, &Output, Options);
@@ -69,7 +68,7 @@ int main(int argc, char *argv[]) {
     clang::clangd::elog("Can not convert response ({0}) to JSON ({1}): {2}\n",
                         Response.DebugString(),
                         (int)JsonStatus.code(),
-                        JsonStatus.message().as_string());
+                        std::string(JsonStatus.message()));
 
     return -1;
   }
